pkgs/by-name/di/disnix-web-service/package.nix

```
@@ -37,9 +46,9 @@ stdenv.mkDerivation (finalAttrs: {
     # add modificationtime="0" to the <jar> and <war> tasks to achieve reproducibility
     xmlstarlet ed -L -a "//jar|//war" -t attr -n "modificationtime" -v "0" build.xml
 
-    sed -i -e "s|#JAVA_HOME=|JAVA_HOME=${jdk}|" \
-       -e "s|#AXIS2_LIB=|AXIS2_LIB=${axis2}/lib|" \
-        scripts/disnix-soap-client
+    substituteInPlace scripts/disnix-soap-client \
+      --replace-fail "#JAVA_HOME=" ${lib.escapeShellArg "JAVA_HOME=${jdk}"} \
+      --replace-fail "#AXIS2_LIB=" ${lib.escapeShellArg "AXIS2_LIB=${finalAttrs.env.AXIS2_LIB}"}
```

As this value is passed to the builder using `env` we can just let Bash do the expansion for us.

```suggestion
      --replace-fail "#AXIS2_LIB=" "AXIS2_LIB=$AXIS2_LIB"
```